packages/expo-modules-autolinking/src/autolinking.ts

```diff
@@ -110,7 +120,69 @@ export async function findModulesAsync(providedOptions: SearchOptions): Promise<
       }
     }
   }
-  return results;
+
+  // It doesn't make much sense to strip modules if there is only one search path.
+  // Workspace root usually doesn't specify all its dependencies (see Expo Go),
+  // so in this case we should link everything.
+  if (options.searchPaths.length <= 1) {
+    return results;
+  }
+  return filterToProjectDependencies(results);
+}
+
+/**
+ * Filters out packages that are not the dependencies of the project.
+ */
+function filterToProjectDependencies(results: SearchResults) {
```

Comment:
cc @EvanBacon - do you think it's worth factoring this out to another module that we can share with config plugins, or are you ok just copy/pasting it over? (for filtering out default config plugins for packages that exist in monorepo but that aren't actually a dep / transitive dep of the project)

Reply:
I'll have to give it some more thought, we can do it pretty easily in the future if needed.